### PR TITLE
NN-3903 remove duplicated adjudications from the database

### PR DESCRIPTION
**Pull request by richardpopple**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/repositories/DraftAdjudicationRepository.kt [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 import org.springframework.data.repository.CrudRepository
 import uk.gov.justice.digital.hmpps.hmppsmanageadjudicationsapi.entities.DraftAdjudication
+import java.time.LocalDateTime
 
 interface DraftAdjudicationRepository : CrudRepository<DraftAdjudication, Long> {
   fun save(draftAdjudication: DraftAdjudication?): DraftAdjudication
 
-  @Query("SELECT da from DraftAdjudication da where da.agencyId = :agencyId and da.createdByUserId = :username and da.reportNumber IS NULL")
-  fun findUnsubmittedByAgencyIdAndCreatedByUserId(agencyId: String, username: String): List<DraftAdjudication>
+  fun findDraftAdjudicationByAgencyIdAndCreatedByUserIdAndReportNumberIsNull(agencyId: String, username: String): List<DraftAdjudication>
```
Comment (richardpopple) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/repositories/DraftAdjudicationRepository.kt:9]: Probably should have done this in a different pull request - apologies. (We prefer the spring naming conventions over annotations?)

File: helm_deploy/hmpps-manage-adjudications-api/templates/housekeeping-cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+apiVersion: batch/v1beta1
```
Comment (richardpopple) [inline helm_deploy/hmpps-manage-adjudications-api/templates/housekeeping-cronjob.yaml:1]: Taken from the sqs stuff - should work but will require a bit of testing in the dev environment most likely

File: helm_deploy/hmpps-manage-adjudications-api/templates/housekeeping-cronjob.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+kind: CronJob
+metadata:
+  name: hmppps-manage-adjudications-api-delete-orphaned
+spec:
+  schedule: "*/10 * * * *"
```
Comment (richardpopple) [inline helm_deploy/hmpps-manage-adjudications-api/templates/housekeeping-cronjob.yaml:6]: every 10 minutes - can change it to something else - but I don't think there is any harm in doing it often.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/DraftAdjudicationService.kt [hunk 4]
```diff
@@ -38,6 +38,11 @@ class DraftAdjudicationService(
   val authenticationFacade: AuthenticationFacade,
 ) {
 
+  @Transactional
+  fun deleteOrphanedDraftAdjudications() {
+    draftAdjudicationRepository.deleteDraftAdjudicationByCreateDateTimeBeforeAndReportNumberIsNotNull(LocalDateTime.now().minusDays(1))
```
[the remark this answers is not in the document]
Comment (sp-watson) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/DraftAdjudicationService.kt:43]: I think that's fine, as these are drafts that are created after someone selected a reported adjudication - so they should have finished with it in a few minutes, never mind a day

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/DraftAdjudicationServiceTest.kt [hunk 5]
```diff
@@ -1041,7 +1041,7 @@ class DraftAdjudicationServiceTest {
   inner class InProgressDraftAdjudications {
     @BeforeEach
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/services/DraftAdjudicationServiceTest.kt:1042]: We ought to have a test for the new method, but it is so simple that I can't see any point.

However, an int test might be good?!

**Review by sp-watson — APPROVED** [review]
